hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/tss/RepeatableTssTests.java

```diff
@@ -0,0 +1,104 @@
+import com.hedera.services.bdd.spec.utilops.streams.assertions.BlockStreamAssertion;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.DynamicTest;
+
+/**
+ * TSS tests that require repeatable mode to run.
+ */
+public class RepeatableTssTests {
+    /**
+     * A test that simulates the behavior of the {@link FakeTssBaseService} under specific conditions
+     * related to signature requests and block creation.
+     *
+     * <p>This test follows three main steps:</p>
+     * <ul>
+     *     <li>Instructs the {@link FakeTssBaseService} to start ignoring signature requests and
+     *     produces several blocks. In this scenario, each transaction is placed into its own round
+     *     since the service is operating in repeatable mode.</li>
+     *     <li>Verifies that no blocks are written, as no block proofs are available, which is the
+     *     expected behavior when the service is ignoring signature requests.</li>
+     *     <li>Reactivates the {@link FakeTssBaseService}, creates another block, and verifies that
+     *     the {@link BlockStreamManager} processes pending block proofs. It checks that the expected
+     *     blocks are written within a brief period after the service resumes normal behavior.</li>
+     * </ul>
+     *
+     * <p>The test ensures that block production halts when block proofs are unavailable and
+     * verifies that the system can catch up on pending proofs when the service resumes.</p>
+     */
+    @RepeatableHapiTest(NEEDS_TSS_CONTROL)
+    Stream<DynamicTest> blockStreamManagerCatchesUpWithIndirectProofs() {
+        final AtomicBoolean expectBlocks = new AtomicBoolean(false);
+        // Once blocks are produced again, the first two will need indirect proofs
+        final AtomicInteger remainingIndirectionLevels = new AtomicInteger(2);
+        final var streamAssertion = new BlockStreamAssertion() {
+            @Override
+            public boolean test(@NonNull final Block block) throws AssertionError {
+                if (!expectBlocks.get()) {
+                    throw new AssertionError("No blocks should be written when TSS is ignoring signature requests");
+                } else {
+                    final var items = block.items();
+                    final var proofItem = items.getLast();
+                    assertTrue(proofItem.hasBlockProof(), "Block proof is expected as the last item");
+                    final var proof = proofItem.blockProofOrThrow();
+                    if (remainingIndirectionLevels.get() == 0) {
+                        assertTrue(
+                                proof.siblingHashes().isEmpty(),
+                                "No sibling hashes should be present on a direct proof");
+                        return true;
+                    } else {
```

Comment:
I wonder about hiding this complexity in a modified StreamFileAlterationListener, one that keeps remainingIndirectionLevels, to make this more approachable. On the other hand, I'm not sure what the consequences of stuffing assertions into a Listener would be yet. Anyway, just a thought.

Reply:
Agreed this isn't readable; extracted to a top-level `IndirectProofsAssertion` that might be reused in places.